iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs

```diff
@@ -639,9 +639,22 @@ public override async Task<TwinProperties> GetTwinAsync(CancellationToken cancel
 
                 if (getTwinResponse.Status != 200)
                 {
-                    throw new IotHubClientException(
-                        getTwinResponse.Message,
-                        IotHubClientErrorCode.NetworkErrors);
+                    // Check if we have an int to string error code translation for the service returned error code.
+                    // The error code could be a part of the service returned error message, or it can be a part of the topic string.
+                    // We will check with the error code in the error message first (if present) since that is the more specific error code returned.
+                    if (Enum.TryParse(getTwinResponse.ErrorResponseMessage.ErrorCode.ToString(), out IotHubClientErrorCode errorCode)
+                        || Enum.TryParse(getTwinResponse.Status.ToString(), out errorCode))
+                    {
+                        throw new IotHubClientException(getTwinResponse.ErrorResponseMessage.Message, errorCode)
```

Comment:
Any idea on whether we can make it retriable or not?

Reply:
Depending on the error code returned IotHubClientException will mark it as transient. That is the only way for us to mark a scenario as retriable.
Note that networking issues won't fit into this scenario as they will not result in a `GetTwinResponse`. This logic only looks at the Status within GetTwinResponse to parse out the error code.

Networking issues will still be handled here:
https://github.com/Azure/azure-iot-sdk-csharp/blob/5508c160d584e77fdf8d3613123d4b9dfbb5767e/iothub/device/src/Transport/Mqtt/MqttTransportHandler.cs#L649-L670